app/components/quiz/QuizContext.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import AppStore from 'store/appStore';
+
+export default React.createClass({
+  displayName: 'QuizContext',
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    context: React.PropTypes.object.isRequired
+  },
+
+  render() {
+    var text = this.props.context.text;
+    var highlights = this.props.context.highlights;
+
+    var start = 0;
+    var tail = '';
+    var l = highlights.length;
+
+    if (highlights[l - 1][1] !== text.length) {
+      tail = <span>{text.substring(highlights[l - 1][1], text.length)}</span>;
+    }
+
+    return (
+      <p className="quiz__context">
+        {highlights.map((n, i) => {
+          // I'd rather not have to wrap these like this... is there a better way?
```

Comment:
oh and if you mean wrapping the span i think that's because react expects a div or a component so wrapping it in parenthesis allows it to wrap this in a div. do you see that in the markup?